### PR TITLE
refactor: improve up ux

### PR DESCRIPTION
**Pull request by evilrobot-01**
Improves the ux of `pop up contract`.

### CODE REVIEW

**Review by AlexD10S — CHANGES_REQUESTED** [review]
Thanks, an improvement over existing UX. 
I just spotted an error that would be great to address in this PR.

File: crates/pop-cli/src/commands/up/contract.rs [hunk 1]
```diff
@@ -63,46 +73,93 @@ pub struct UpContractCommand {
...
-		if !is_chain_alive(self.url.clone()).await? {
+		// Check if specified chain is accessible
+		let process = if !is_chain_alive(self.url.clone()).await? {
+			let url = self.url.as_str();
+
 			if !self.skip_confirm {
+				let chain = if self.url.as_str() == DEFAULT_URL {
+					"No endpoint was specified.".into()
+				} else {
+					format!("The endpoint at \"{url}\" is not accessible.")
```
[the remark this answers is not in the document]
Comment (evilrobot-01) [reply crates/pop-cli/src/commands/up/contract.rs:108]: I don't follow. Do you mean if one specifies a URL to some endpoint and if it isn't available, launching the local node is pointless as the the same invalid URL is used?

[the remark this answers is not in the document]
Comment (AlexD10S) [reply crates/pop-cli/src/commands/up/contract.rs:108]: The issue is that `pop` displays a message indicating that the selected endpoint (e.g., `ws://localhost:9333`) is not accessible. It then prompts the user to start a local node for testing. If the user agrees, the system starts the `substrate-contracts-node` on `ws://localhost:9944`, but still calls the upload function to deploy into the originally provided URL (`ws://localhost:9333`), which fails. You can test with `pop up contract --args false  --url ws://localhost:9333`

[the remark this answers is not in the document]
Comment (evilrobot-01) [reply crates/pop-cli/src/commands/up/contract.rs]: Thanks for clarifying. I have addressed by simply updating the url to the default after user confirmation, which then works fine using your sample command.

Parsing ports and using it to pass as a parameter to s-c-n doesnt necessarily make sense, as the url specified might be an actual network - e.g. `--url wss://rpc2.paseo.popnetwork.xyz`. We could look at the components of the url to see if it is localhost to try and use the port specified, but think that can be added separately if required.

**Review by AlexD10S — APPROVED** [review]

**Pull request merged**